src/framework/theme/components/spinner/spinner.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, HostBinding, Input } from '@angular/core';
+import { NbCardComponent } from '../card/card.component';
+
+@Component({
+  selector: 'nb-spinner',
+  template: `
+    <span class="loading  loading-{{spinnerSize}}">
+      <span class="set-height">d</span>
+      <span class="spin-circle"></span>
+      <span class="loading-text">{{text}}</span>
+    </span>
+  `,
+})
+
+export class NbSpinnerComponent {
+
+  static readonly SIZE_SMALL = 'small';
+  static readonly SIZE_MEDIUM = 'medium';
+  static readonly SIZE_LARGE = 'large';
+
+
+  static readonly STATUS_PRIMARY = 'primary';
+  static readonly STATUS_INFO = 'info';
+  static readonly STATUS_SUCCESS = 'success';
+  static readonly STATUS_WARNING = 'warning';
+  static readonly STATUS_DANGER = 'danger';
+
+
+  @Input() visible: boolean;
+  spinnerSize: string;
+
+  @Input('size') set size(val: string) {
+    if (val === NbSpinnerComponent.SIZE_SMALL || NbSpinnerComponent.SIZE_MEDIUM ||
+      NbSpinnerComponent.SIZE_LARGE) {
+      this.spinnerSize = val
+    } else {
+      this.spinnerSize = NbSpinnerComponent.SIZE_MEDIUM
+    }
+  };
+
+  @Input() disableBackground: boolean;
```

Comment:
what does this property mean?
if I understand correctly `noBackground` or `hideBackground` would be a better name